### PR TITLE
Configuration/Arguments refactoing (don't pass Namespace around).

### PR DESCRIPTION
**Pull request by xmatthias**
## Summary
Simplify configuration (passing a dict instead of `Namespace` to it).
Where possible, we can use `args.get('argument')` instead of having to check everything first.

This will allow simpler (and correct) initialization of configurations in interactive environments (still using `Configuration.from_files([files...])`).

This should also solve #2078, since a simple (sample) configuration can be passed to `from_files()` and will be correctly initialized (including the correct types for data_dir and user_data_dir) - as pointed out in #2228 .

## Quick changelog

- Use vars(args) to avoid passing around Namespace objects
- Refactor `from_files()` to correctly initialize configuration 
- Add create-datadir to the commands that don't require config to be set.

### CODE REVIEW

File: docs/data-analysis.md [hunk 1]
```diff
@@ -101,7 +102,16 @@ from freqtrade.configuration import Configuration
 config = Configuration.from_files(["config1.json", "config2.json"])
 
 # Show the config in memory
-print(json.dumps(config, indent=1))
+print(json.dumps(config, indent=2))
+```
+
+For Interactive environments, have an additional configuration specifying `user_data_dir` and pass this in last, so you don't have to change directories while running the bot.
+Best avoid relative paths, since this starts at the storage location of the jupyter notebook, unless the folder is changed.
```
Comment (hroff-1902) [inline docs/data-analysis.md]: No folders, please! 
🤣

**Comment by hroff-1902** [comment]
LGTM, besides a comment above on the use of the term for directories, and a suggestion, a wish below:

if it's possible to run quickly an auto replacement on the changes made here, let's move all `args["argname"]` to `args['argname']`. Although it's not in PEPs and in the Zen of python, this could later help during hypothetical internationalization, resolving I18N issues. I mean, keeping as a rule, that the 'translatable' messages printed to the humans are in the double quotes, while intrinsic string constants, dict keys, options, settings are written in the single quotes. I do not insist on this (but believe me, it can help during I18N processes), and this is only if it can be done with little effort...

**Comment by xmatthias** [comment]
> if it's possible to run quickly an auto replacement on the changes made here, let's move all args["argname"] to args['argname']. Although it's not in PEPs and in the Zen of python, this could later help during hypothetical internationalization, resolving I18N issues. I mean, keeping as a rule, that the 'translatable' messages printed to the humans are in the double quotes, while intrinsic string constants, dict keys, options, settings are written in the single quotes. I do not insist on this (but believe me, it can help during I18N processes), and this is only if it can be done with little effort...

While certainly possible, i am not sure how that could help with internationalization - I guess you're thinking about auto-replacing stuff (?) - however you'll need to evaluate that for every string anyway, to make sure you're not replacing something wrongly.
Never done I18N with python, but i assume you'll have a dict (or other container) for each string to print and you select from that (probably thorugh a library...).

However, I don't see a point in internationalization for freqtrade - it's about trading, not about accessibility - getting a log-message in another language does not give anyone any more profits, so i think that'll be wasted time (we can think about this again should we have an UI ...). - and immagine the problems when we get issues with logs in languages noone of us understands ... 

Regarding formatting of strings - I'm more drawn towards [black](https://black.readthedocs.io/en/stable/) - which formats the python code in a certain way (strings are in doublequotes unless that requires escaping doublequotes within the string, line-breaks are standardized, ...).

We're a long way from that (running black modifies every file in the project ...) - however what i like about it is that it'll terminate discussions around formatting - since black is THE standardized format.

**Comment by hroff-1902** [comment]
Citing from black manual on strings:

> (like the popular “single quotes for data, double quotes for human-readable strings”)

😉 

Okay, black'ish style is also okay. Forget that above.

**Pull request merged**